### DISCUSSION TITLE
Can I use Vivado 24

### DISCUSSION THREAD

**Discussion by DrBwts** [fastmachinelearning/hls4ml#1057] [Q&A]
I see the recommendations are Vivado  HLS versions between 2018.2 and 2020.1 but will HLS4ML still work with Vivado 2024.1?

**Comment by JanFSchulte** [comment]
What was used to be called Vivado HLS is now Vitis HLS, so for newer versions of the Xilinx tools, you can use the Vitis backend in hls4ml. If I understand the current product naming correctly, Vivado 2024.1 does not support HLS but is rather meant for HDL design. So Vitis HLS is the tool to use for work with hsl4ml.

**Comment by jmitrevs** [comment]
And just to be clear, you can use the IP produced with Vitis HLS in Vivado. It's just that Vivado HLS no longer exists.

**Comment by DrBwts** [comment]
So I'll be OK using the latest integrated Vivado/Vitis 2024 install?

**Comment by JanFSchulte** [comment]
Yes, if you use the Vitis backend in hls4ml, that should work. I'm not 100% sure how much we have tested the latest Vitis version, though, maybe @jmitrevs and @vloncar can comment on that.

**Comment by jmitrevs** [comment]
I personally haven't used 2024.x much, but I would expect it to work. You should use the main branch of hls4ml, though, not the tagged release. Vitis HLS 2022.2 and later should be supported.
